packages/web-components/fast-html/src/components/element.ts

```diff
@@ -1,35 +1,19 @@
-import { attr, FASTElement, Observable } from "@microsoft/fast-element";
+import { attr, FASTElement } from "@microsoft/fast-element";
 
 export abstract class RenderableFASTElement extends FASTElement {
     @attr({ mode: "boolean", attribute: "defer-hydration" })
     deferHydration: boolean = true;
 
-    @attr({ mode: "boolean", attribute: "needs-hydration" })
-    needsHydration: boolean = true;
-
     constructor() {
         super();
 
-        // If the elements template has already been defined there is no need to defer hydration.
-        if (!this.$fastController?.definition?.shadowOptions) {
-            this.setAttribute("defer-hydration", "");
-            this.setAttribute("needs-hydration", "");
+        if (this.prepare) {
+            this.prepare().then(() => {
+                this.deferHydration = false;
+            });
+        } else {
+            this.deferHydration = false;
         }
```

Comment:
nit: I think this could be expressed as:
```ts
this.prepare?.().then(() => {
    this.deferHydration = false;
}) ?? (this.deferHydration = false);
```
...but our tslib configuration would likely expand it to be longer than what you already have here.

Reply:
Let's keep it simple here then and revisit updating tslib and typescript for v3, there's probably quite a bit of clean up to evaluate on that change alone.